### ISSUE TITLE
Volume is set to half of requested level (0–100 scale)

### ISSUE THREAD

**Issue by lukestein** [DoctorNSA/homebridge-musiccast-tv#15]
Thank you for the great plugin!

My fan control runs on a 0–100% "speed" scale, but sets the volume on my TSR-700 to half the chosen level; e.g., if I choose 75% using the fan controller, my AVR will be set to a volume value of 37.5 on its displayed 0–100 scale.

There appears to be no way to set the AVR's volume above 50/100 using the fan controller.

I am not sure if this is related, but I also regularly get this notice in my homebridge logs:

> [12/04/2021, 16:27:57] [homebridge-musiccast-tv] This plugin generated a warning from the characteristic 'Volume': characteristic was supplied illegal value: number 161 exceeded maximum of 100. See https://git.io/JtMGR for more info.

I have tried to set the `maxVol` to various values (e.g., 80, 100, 161) with no apparent effect on either the illegal value notice, or my ability to set volumes above 50.

In case useful, here's the (prettified json) I get at `/YamahaExtendedControl/v1/system/getFeatures`:

```json
{
  "response_code": 0,
  "system": {
    "func_list": [
      "wired_lan",
      "wireless_lan",
      "network_standby",
      "network_standby_auto",
      "bluetooth_standby",
      "bluetooth_tx_setting",
      "bluetooth_tx_connectivity_type",
      "dfs_option",
      "headphone",
      "hdmi_out_1",
      "airplay",
      "disklavier_settings",
      "background_download",
      "remote_info",
      "network_reboot",
      "system_reboot",
      "ypao_volume",
      "party_mode",
      "party_volume",
      "party_mute",
      "name_text_avr",
      "hdmi_standby_through",
      "analytics"
    ],
    "zone_num": 2,
    "input_list": [
      {
        "id": "napster",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": true,
        "play_info_type": "netusb"
      },
      {
        "id": "siriusxm",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": true,
        "play_info_type": "netusb"
      },
      {
        "id": "pandora",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": true,
        "play_info_type": "netusb"
      },
      {
        "id": "spotify",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": false,
        "play_info_type": "netusb"
      },
      {
        "id": "qobuz",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": true,
        "play_info_type": "netusb"
      },
      {
        "id": "tidal",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": true,
        "play_info_type": "netusb"
      },
      {
        "id": "deezer",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": true,
        "play_info_type": "netusb"
      },
      {
        "id": "amazon_music",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": true,
        "play_info_type": "netusb"
      },
      {
        "id": "alexa",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": false,
        "play_info_type": "netusb"
      },
      {
        "id": "airplay",
        "distribution_enable": false,
        "rename_enable": false,
        "account_enable": false,
        "play_info_type": "netusb"
      },
      {
        "id": "mc_link",
        "distribution_enable": false,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "netusb"
      },
      {
        "id": "server",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "netusb"
      },
      {
        "id": "net_radio",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "netusb"
      },
      {
        "id": "bluetooth",
        "distribution_enable": true,
        "rename_enable": false,
        "account_enable": false,
        "play_info_type": "netusb"
      },
      {
        "id": "usb",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "netusb"
      },
      {
        "id": "tuner",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "tuner"
      },
      {
        "id": "hdmi1",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "hdmi2",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "hdmi3",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "hdmi4",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "hdmi5",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "hdmi6",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "hdmi7",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "tv",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "audio1",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "audio2",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "audio3",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "audio4",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "audio5",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "phono",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      },
      {
        "id": "main_sync",
        "distribution_enable": true,
        "rename_enable": true,
        "account_enable": false,
        "play_info_type": "none"
      }
    ],
    "bluetooth": {
      "update_cancelable": true
    },
    "web_control_url": "/AdvancedSettings.html?mc=true",
    "party_volume_list": [
      "up",
      "down"
    ],
    "hdmi_standby_through_list": [
      "off",
      "on",
      "auto"
    ]
  },
  "zone": [
    {
      "id": "main",
      "func_list": [
        "power",
        "sleep",
        "volume",
        "mute",
        "sound_program",
        "pure_direct",
        "enhancer",
        "tone_control",
        "dialogue_level",
        "dialogue_lift",
        "subwoofer_volume",
        "signal_info",
        "prepare_input_change",
        "link_control",
        "link_audio_delay",
        "scene",
        "contents_display",
        "cursor",
        "menu",
        "actual_volume",
        "surr_decoder_type",
        "extra_bass",
        "adaptive_drc",
        "dts_dialogue_control",
        "adaptive_dsp_level"
      ],
      "input_list": [
        "napster",
        "siriusxm",
        "pandora",
        "spotify",
        "qobuz",
        "tidal",
        "deezer",
        "amazon_music",
        "alexa",
        "airplay",
        "mc_link",
        "server",
        "net_radio",
        "bluetooth",
        "usb",
        "tuner",
        "hdmi1",
        "hdmi2",
        "hdmi3",
        "hdmi4",
        "hdmi5",
        "hdmi6",
        "hdmi7",
        "tv",
        "audio1",
        "audio2",
        "audio3",
        "audio4",
        "audio5",
        "phono"
      ],
      "sound_program_list": [
        "munich",
        "vienna",
        "chamber",
        "cellar_club",
        "roxy_theatre",
        "bottom_line",
        "sports",
        "action_game",
        "roleplaying_game",
        "music_video",
        "standard",
        "spectacle",
        "sci-fi",
        "adventure",
        "drama",
        "mono_movie",
        "2ch_stereo",
        "all_ch_stereo",
        "surr_decoder",
        "straight"
      ],
      "surr_decoder_type_list": [
        "toggle",
        "auto",
        "dolby_surround",
        "dts_neural_x",
        "dts_neo6_cinema",
        "dts_neo6_music"
      ],
      "tone_control_mode_list": [
        "manual"
      ],
      "link_control_list": [
        "speed",
        "standard",
        "stability"
      ],
      "link_audio_delay_list": [
        "audio_sync",
        "balanced",
        "lip_sync"
      ],
      "range_step": [
        {
          "id": "volume",
          "min": 0,
          "max": 161,
          "step": 1
        },
        {
          "id": "tone_control",
          "min": -12,
          "max": 12,
          "step": 1
        },
        {
          "id": "dialogue_level",
          "min": 0,
          "max": 3,
          "step": 1
        },
        {
          "id": "dialogue_lift",
          "min": 0,
          "max": 5,
          "step": 1
        },
        {
          "id": "subwoofer_volume",
          "min": -12,
          "max": 12,
          "step": 1
        },
        {
          "id": "actual_volume_db",
          "min": -80.5,
          "max": 16.5,
          "step": 0.5
        },
        {
          "id": "actual_volume_numeric",
          "min": 0.0,
          "max": 97.0,
          "step": 0.5
        },
        {
          "id": "dts_dialogue_control",
          "min": 0,
          "max": 6,
          "step": 1
        }
      ],
      "scene_num": 8,
      "cursor_list": [
        "up",
        "down",
        "left",
        "right",
        "select",
        "return"
      ],
      "menu_list": [
        "on_screen",
        "top_menu",
        "menu",
        "option",
        "display",
        "help",
        "home",
        "mode",
        "red",
        "green",
        "yellow",
        "blue"
      ],
      "actual_volume_mode_list": [
        "db",
        "numeric"
      ],
      "ccs_supported": [
        "connected_speaker"
      ]
    },
    {
      "id": "zone2",
      "zone_b": false,
      "func_list": [
        "power",
        "sleep",
        "volume",
        "mute",
        "enhancer",
        "tone_control",
        "balance",
        "prepare_input_change",
        "link_control",
        "scene",
        "actual_volume",
        "extra_bass",
        "mono"
      ],
      "input_list": [
        "napster",
        "siriusxm",
        "pandora",
        "spotify",
        "qobuz",
        "tidal",
        "deezer",
        "amazon_music",
        "alexa",
        "airplay",
        "mc_link",
        "server",
        "net_radio",
        "bluetooth",
        "usb",
        "tuner",
        "audio1",
        "audio2",
        "audio3",
        "audio4",
        "audio5",
        "phono",
        "main_sync"
      ],
      "tone_control_mode_list": [
        "manual",
        "auto",
        "bypass"
      ],
      "link_control_list": [
        "speed",
        "standard",
        "stability"
      ],
      "range_step": [
        {
          "id": "volume",
          "min": 0,
          "max": 161,
          "step": 1
        },
        {
          "id": "tone_control",
          "min": -12,
          "max": 12,
          "step": 1
        },
        {
          "id": "balance",
          "min": -20,
          "max": 20,
          "step": 1
        },
        {
          "id": "actual_volume_db",
          "min": -80.5,
          "max": 10.0,
          "step": 0.5
        },
        {
          "id": "actual_volume_numeric",
          "min": 0.0,
          "max": 90.5,
          "step": 0.5
        }
      ],
      "scene_num": 8,
      "actual_volume_mode_list": [
        "db",
        "numeric"
      ]
    }
  ],
  "tuner": {
    "func_list": [
      "am",
      "fm"
    ],
    "range_step": [
      {
        "id": "am",
        "min": 530,
        "max": 1710,
        "step": 10
      },
      {
        "id": "fm",
        "min": 87500,
        "max": 107900,
        "step": 200
      }
    ],
    "preset": {
      "type": "common",
      "num": 40
    }
  },
  "netusb": {
    "func_list": [
      "recent_info",
      "play_queue",
      "mc_playlist",
      "streaming_service_use"
    ],
    "preset": {
      "num": 40
    },
    "recent_info": {
      "num": 40
    },
    "play_queue": {
      "size": 200
    },
    "mc_playlist": {
      "size": 200,
      "num": 5
    },
    "net_radio_type": "airable",
    "pandora": {
      "sort_option_list": [
        "recent",
        "alphabet"
      ]
    },
    "siriusxm": {
      "api_type": "everest"
    }
  },
  "distribution": {
    "version": 2.0,
    "compatible_client": [
      2
    ],
    "client_max": 19,
    "server_zone_list": [
      "main",
      "zone2"
    ],
    "mc_surround": {
      "version": 1.0,
      "func_list": [
        "mcss_error_info"
      ],
      "master_role": {
        "surround_pair": true,
        "stereo_pair": false,
        "subwoofer_pair": true
      },
      "slave_role": {
        "surround_pair_l_or_r": false,
        "surround_pair_lr": false,
        "subwoofer_pair": false
      }
    }
  },
  "ccs": {
    "supported": true
  }
}
```

**Comment by aangert** [comment]
I have found the bug and will fix it in the next update

**Comment by aangert** [comment]
Sorry I forgot about this issue, a fix will be published in the next days

**Comment by aangert** [comment]
Fixed in v1.7.1
